### PR TITLE
Add courses to the repository

### PR DESCRIPTION
**Pull request by statnmap**
The content will be built using an internal ThinkR package called {formation}.
This ensures that the HTML output, datasets, and extra files are stored in the correct place to be added to the ThinkR e-learning platform.  
Each instructor is still able to locally knit each chapter using {xaringan}.  
The content of the course is:  

- courses/C00-package-objectives.en.Rmd: What we will present during this tutorial, and how it will be held
- courses/00-cruz.Rmd: How the e-learning platform works
- courses/C00-package_preambule.en.Rmd: What means package first ?
- courses/C01-package_express_fusen.en.Rmd: Build a package using {fusen}
- courses/C06-pkg_avance_test.Rmd: Go further with unit tests

### CODE REVIEW

**Comment by statnmap** [comment]
I plan to try a GitHub Actions to compile the course with the ThinkR template.

File: courses/C01-package_express.en.Rmd [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
++ Redo this package {hello2} on your own in a new project
+
+
+---
+class: slide 
+### Step 1: Create a new "package" project
+
+In Rstudio : 
+
+File > New project > new directory > R package **using devtools**
```
Comment (emilyriederer) [inline courses/C01-package_express.en.Rmd]: Should we use `usethis` to create the directory instead of the IDE?

File: courses/C01-package_express.en.Rmd [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+---
+class: slide 
+### Step 1: Create a new "package" project
+
+In Rstudio : 
+
+File > New project > new directory > R package **using devtools**  
+We choose the name of the package (explicit, in lower case) and a folder on the computer  
+
+- Name of the package : "hello
```
Comment (emilyriederer) [inline courses/C01-package_express.en.Rmd]: ^Missing close quotation mark

File: courses/C01-package_express.en.Rmd [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+> _no capital letters, periods, underscores or spaces_.
+
+---
+class: slide 
+### Step 2 : Description
+
+The description of the package is done in the "DESCRIPTION" file.  
+The first fields to fill in :
+
+- Title: quick description
```
Comment (emilyriederer) [inline courses/C01-package_express.en.Rmd]: I know CRAN has some strict rules around title/description (e.g. don't reuse package name, certain capitalization, etc.) Even though most _analyisis_ packages will not end up on CRAN should we still provide these as the best practices?

File: courses/C01-package_express.en.Rmd [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+- Name of the package : "hello
+
+> _no capital letters, periods, underscores or spaces_.
+
+---
+class: slide 
+### Step 2 : Description
+
+The description of the package is done in the "DESCRIPTION" file.  
+The first fields to fill in :
```
Comment (emilyriederer) [inline courses/C01-package_express.en.Rmd]: Do we wish to encourage participants to edit the `DESCRIPTION` directly or via the `usethis` functions? I have no strong preference

File: courses/C01-package_express.en.Rmd [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+> The rest, we'll see later_.
+
+---
+class: slide 
+
+### Step 3: dev_history.R
+
+We create a file to save all our manual operations
+
+- Create a "dev/" folder
```
Comment (emilyriederer) [inline courses/C01-package_express.en.Rmd]: I see that you use `usethis::use_build_ignore()` to ignore the `dev/` folder. I wonder if this is worth explicitly listing out as a step and explaining that _any_ additional files/directories can be included in a package but those not related to the core package must be ignored?

File: courses/C01-package_express.en.Rmd [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+library(hello)
+```
+
+## Say hello to someone
+
+`r ''````{r}
+print("Hello person")
+
+person <- "Seb"
+print(glue::glue("Hello {person}"))
```
Comment (emilyriederer) [inline courses/C01-package_express.en.Rmd]: When you've taught this previously, does everyone seem to know the `glue` package? I think it is pretty obvious what this code is doing (which is a good thing!), but alternatively we could use `paste()` since everyone is familiar

Comment (emilyriederer) [reply courses/C01-package_express.en.Rmd]: However, I do see later that this motivates the use of `importFrom` in the Rd, so perhaps this is strategic! 🙂

File: courses/C06-pkg_avance_test.Rmd [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+Your package must be documented, **tested** and return 0 errors, 0 warnings, and 0 notes
+
+---
+class: slide
+
+### Hands-on
+
+- Add unit test in your {hello} package to test that `say_hello()` returns a message
+- Check the package
+
```
Comment (emilyriederer) [inline courses/C06-pkg_avance_test.Rmd]: I wonder if we could/should add anything about how to come up with good test cases? I often think new developers do not think carefully about creating _hypotheses_ for how their code could fail and evaluating actual corner cases

File: courses/C01-package_express_fusen.en.Rmd [hunk 8]
```diff
@@ -0,0 +1,540 @@
...
+`r ''````{r tests}
+test_that("say_hello works", {
+  expect_message(say_hello(someone = "Seb"), "Hello Seb")
+})
+```
+````
+
+???
+
+- We use the same exact example to rite the unit test directly.
```
Comment (emilyriederer) [inline courses/C01-package_express_fusen.en.Rmd]: Tiny typo: "rite" -> "write"

File: courses/C01-package_express_fusen.en.Rmd [hunk 9]
```diff
@@ -0,0 +1,540 @@
...
+- Doing _version control_ with _git
+  + `usethis::use_git()`
+  
+- Add the README to your git repository
+  + `usethis::use_readme_rmd()`
+  
+- Use a test set for your examples and tests
+  + `usethis::use_data()`: function to create an internal dataset  
+  
+- Add a version tracking file
```
Comment (emilyriederer) [inline courses/C01-package_express_fusen.en.Rmd:782]: Should we also mention `usethis::use_version()`? Participants coming from an analysis context might also appreciate advise on when/how often to update different parts of the version number

**Review by emilyriederer — COMMENTED** [review]
Thank you for sharing -- this looks great! I added a few minor thoughts/comments as I read through just for discussion, but there is nothing I feel strongly about changing. 

I have one broader question about `fusen`: once one "inflates" the package, do they need to continue to develop exclusively within the Rmd or exclusively from the generated files? I am wondering if we should teach any changes within the resulting package file structure or if the two workflows are not compatible.

File: courses/C01-package_express.en.Rmd (hunk 1 above)
Comment (statnmap) [reply courses/C01-package_express.en.Rmd]: I add the possibility. Thanks.

File: courses/C01-package_express.en.Rmd (hunk 2 above)
Comment (statnmap) [reply courses/C01-package_express.en.Rmd]: Corrected in the next version

File: courses/C01-package_express.en.Rmd (hunk 3 above)
Comment (statnmap) [reply courses/C01-package_express.en.Rmd]: Yes. Some of these rules can break the check(). Added.

File: courses/C01-package_express.en.Rmd (hunk 6 above)
Comment (statnmap) [reply courses/C01-package_express.en.Rmd]: Yes this is strategic :smile:

File: courses/C01-package_express_fusen.en.Rmd (hunk 9 above)
Comment (statnmap) [reply courses/C01-package_express_fusen.en.Rmd:782]: That is a common question yes. Let's add this.

**Comment by statnmap** [comment]
I removed the course named "courses/C01-package_express.en.Rmd", I think we can directly use the version with {fusen} (courses/C01-package_express_fusen.en.Rmd)  
I updated its content following your feedbacks and those of my current attendees.  
I removed the course on unit tests, as we proposed to speak about including data in the package. I do not think we will be able to cover both topics.  
Then, I added a course on data, still in French for now, but I will translate it, and add an example of use from {fusen} using the full dev template with `fusen::add_dev_history()`
